config/api/src/main/java/org/jboss/windup/config/Variables.java

```diff
@@ -199,21 +201,24 @@ public void removeVariable(String name)
      * Searches the variables layers, top to bottom, for the iterable having all of it's items of the given type. Return
      * null if not found.
      */
-    public Iterable<? extends WindupVertexFrame> findVariableOfType(Class<?> type)
+    public <T> Iterable<T> findVariableOfType(Class<T> type)
```

Comment:
I don't quite understand what you mean. From what I understood the answer would be that it returns only the Frames, because only Frames are saved in the Variables stack. And since it looks for only the frames that are of specific type, we are sure that the result will be the type that is searched for, so why not to cast it?